### ISSUE TITLE
k3.jar is not portable

### ISSUE THREAD

**Issue by osa1** [kframework/k-legacy#129]
Just realized that k3.jar is not portable:

```
➜  ~  java -jar k3.jar -krun
Exception in thread "main" java.lang.NoClassDefFoundError: jline/Completor
        at org.kframework.main.Main.main(Main.java:59)
Caused by: java.lang.ClassNotFoundException: jline.Completor
        at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:423)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:356)
        ... 1 more
```

is this by design or bug? Personally, I'd like to have portable jar file that I could simply scp to my remote machine and run. (this should also make distributing easier)

**Comment by radumereuta** [comment]
Define portable. I'm not sure what happened there.
We should definitely have portability, in the sense that if you take the jar from one environment and you put it into another K environment, it should work. But k3.jar should not work on it's own. It needs all the other tools, like sdf2table, the /include directory, /lib. It needs to be in exactly the right place in the tool.
Also, we want to keep the release functionality. Meaning that we want to export one archive that contains the entire tool, compiled and ready to go. As long that we have this, I think we are safe.

Can you explain in a bit more details what you did that you got that error?

**Comment by osa1** [comment]
I just copied k3.jar to another place on same machine and it failed to run. It's easy to reproduce, just move k3.jar to somewhere else.

> But k3.jar should not work on it's own. It needs all the other tools, like sdf2table, the /include directory, /lib. It needs to be in exactly the right place in the tool.

I was expecting this dependencies to be copied to jar.

> Also, we want to keep the release functionality. Meaning that we want to export one archive that contains the entire tool, compiled and ready to go. As long that we have this, I think we are safe.

Then we're not safe because k3 archive file(jar) doesn't contain all dependencies.

**Comment by radumereuta** [comment]
It's not even supposed to contain all dependencies. We're not exporting a java project. We are shipping an entire framework with all kind of additional stuff. We need the different scripts to be able to run it on all platforms. We need the executables. In this case we don't want to include them in the jar, because it would not be very efficient. You need to run them so many times... You need to be able to see the examples.

In other words, we don't ship only k3.jar. Our product is more complex and that jar is only a small part of the tool.
